scripts/Test-RepoPackage.ps1

```diff
@@ -0,0 +1,633 @@
+  }
+  end {
+    $queries
+   }
+}
+
+function GetPackagesFromDiff() {
+<#
+  Gets the changed packages that have changed since the
+  latest common ancestor of the current branch, and the specified $diffAgainst branch.
+
+.PARAMETER diffAgainst
+  The branch we should run git diff against.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNull()]
+    [string]$diffAgainst
+  )
+  $paths = git diff "${diffAgainst}..." --name-only | % {
+    if ($_.StartsWith('..')) {
+      $path = Split-Path -Parent $_
+    } else {
+      $path = Split-Path -Parent (Resolve-Path -Relative "$PSScriptRoot\..\$_")
+    }
+    while ($path -and !(Test-Path "$path\*.nuspec")) {
+      $path = Split-Path -Parent $path
+    }
+    if ($path) { $path }
+  } | select -Unique
+
+  return $paths | GetPackageSelectQuery
+}
+
+function GetPackagesFromName() {
+  param(
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [string]$packageName
+  )
+  return Get-ChildItem "$PSScriptRoot\..\" -Recurse -Filter "$packageName.nuspec" | select -Unique -expand Directory | GetPackageSelectQuery
+}
+
+function MoveLogFile() {
+<#
+.SYNOPSIS
+  Renames the chocolatey.log file to a file matching the specified packageName and commandType
+  so it only contains the necessary log for the current package.
+#>
+  param(
+    [Parameter(Mandatory = $true)]
+    [string]$packageName,
+    [Parameter(Mandatory = $true)]
+    [string]$commandType
+  )
+
+  if (Test-Path "$env:ChocolateyInstall\logs\chocolatey.log") {
+    mv "$env:ChocolateyInstall\logs\chocolatey.log" "$env:ChocolateyInstall\logs\${commandType}_${packageName}.log" | WriteOutput
+  }
+}
+
+function RemoveDependentPackages() {
+  param(
+    [object]$packages
+  )
+
+  [array]$dependentPackages = $packages | ? { $_.DependentPackage -ne $null -and $_.DependentPackage -ne '' } | select -expand DependentPackage
+  if ($dependentPackages -ne $null -and $dependentPackages.Count -gt 0) {
+    $packages = $packages | ? { !$dependentPackages.Contains($_.Name) }
+  }
+  return $packages
+}
+
+function RunChocoPackProcess() {
+<#
+.SYNOPSIS
+  Function responsible for running choco pack when a nupkg file have not already been created.
+#>
+  param([string]$path)
+  if ($path -ne $null -and $path -ne '') { pushd $path }
+  if (!(Test-Path "*.nupkg")) {
+    . choco pack | WriteChocoOutput
+    if ($LastExitCode -ne 0) { popd; throw "Choco pack failed with code: $LastExitCode"; return }
+  }
+  if ($path -ne $null -and $path -ne '') { popd}
+}
+
+function SetAppveyorExitCode() {
+<#
+.SYNOPSIS
+  Sets the exit code of the script when running on appveyor, so
+  the build would fail when necessary.
+#>
+  param(
+    [int]$ExitCode
+  )
+  WriteOutput "Exit code was $ExitCode" -type Warning
+
+  if (!(Test-Path env:\APPVEYOR)) {
+    return
+  }
+  if ($ExitCode -ne 0) {
+    $host.SetShouldExit($ExitCode)
+  }
+}
+
+function UploadAppveyorArtifact() {
```

Comment:
The script is meant to be run on appveyor, with that said though.
This function doesn't do anything if the `$env:APPVEYOR` haven't been set.
So it's possible to run it on a local machine as well.